src/scanner.c

```
@@ -0,0 +1,318 @@
+
+enum TokenType {
+    HEREDOC_MARKER,
+    HEREDOC_LINE,
+    HEREDOC_END,
+    HEREDOC_NL,
+    ERROR_SENTINEL,
+};
+
+void *tree_sitter_containerfile_external_scanner_create() {
+    scanner_state *state = malloc(sizeof(scanner_state));
+    memset(state, 0, sizeof(scanner_state));
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)